iothub/device/src/Transport/Amqp/AmqpConnectionHolder.cs

```diff
@@ -73,6 +73,7 @@ public void Shutdown()
 
             _amqpAuthenticationRefresher?.StopLoop();
             _amqpIotConnection?.SafeClose();
+            _amqpIotConnection.Closed -= OnConnectionClosed;
```

Comment:
you'll want to decrement the event handler before you make an explicit close call. Otherwise, the close call will end up invoking the handler (which is not required in this scenario)

Reply:
updated